src/entities/DealTokenSwap.ts

```diff
@@ -204,6 +209,33 @@ export class DealTokenSwap implements IDeal {
     return this.fundingWasInitiated && !this.isExecuted && !this.fundingPeriodHasExpired;
   }
 
+  /**
+   * Gets the DAO based on the connected wallet address
+   */
+  @computedFrom("ethereumService.defaultAccountAddress")
+  public get daoRelatedToWallet(): IDAO{
+    return this.getDao(true);
+  }
+
+  /**
+    * Gets the non-related DAO based on the connected wallet address
+    */
+  @computedFrom("ethereumService.defaultAccountAddress")
+  public get otherDao(): IDAO {
+    return this.getDao(false);
+  }
+
+  /**
+    * Gets the DAO based on the connected wallet address
+    */
+  private getDao(relatedToWallet: boolean) : IDAO {
+    if (this.registrationData.partnerDAO.representatives.some(x => x.address === this.ethereumService.defaultAccountAddress)){
+      //the connected wallet is a representative of the partner DAO
+      return relatedToWallet ? this.registrationData.partnerDAO : this.registrationData.primaryDAO;
```

Comment:
Currently, the calling code is verifying that the connected wallet is at least part of one of the daos or else it will be redirected to the home page. I wasn't done fleshing this part out yet though because when I wrote it, I wasn't 100% sure which DAO the proposal lead would be for.

Reply:
I changed it to check for both and return null if the current wallet address isn't in either. Also if the current wallet isn't in either and you're the proposal lead I just made the primaryDao the DAO related to the wallet.